### PR TITLE
feat: create tasks to process a manual and automatic amendment request. + bug fixes

### PR DESCRIPTION
**Pull request by IgnatG**
#### Features:
- Create tasks to process an automatic amendment request (DTFS2-5681)
- Create tasks to process a manual amendment request (DTFS2-4608)
- Warning text on the task list page displays "facility (facility ID number)" for each facility being amended (DTFS2-5866)
- Display multiple amendments within the task list tab (DTFS2-5864)

#### Bug fixes:
- TFM - Underwriting tab - Underwriter should be able to change lead underwriter (DTFS2-5815)
- TFM - Underwriting tab - Lead underwriters email address should be next to the name (DTFS2-5814)
- Amendments removed from underwriting tab once bank's decision is added (DTFS2-5858)

### CODE REVIEW

File: trade-finance-manager-ui/server/test-mocks/amendment-test-mocks.js [hunk 1]
```diff
@@ -45,6 +45,28 @@ const session = {
...
+  requireUkefApproval: true,
+};
+
+const MOCK_AMENDMENT_AUTOMATIC_APPROVAL = {
+  amendmentId: '12345',
+  facilityId: '45678',
+  dealId: '999',
+  ukefFacilityId: '123',
+  type: CONSTANTS.FACILITY.FACILITY_TYPE.CASH,
+  status: CONSTANTS.AMENDMENTS.AMENDMENT_STATUS.IN_PROGRESS,
```
Comment (IgnatG) [inline trade-finance-manager-ui/server/test-mocks/amendment-test-mocks.js:64]: if it's submitted by PIM and it does not require UKEF approval, then the status should be `Completed`
Also, can we inherit the properties from `MOCK_AMENDMENT` and override anything that's different?

File: trade-finance-manager-ui/server/controllers/case/index.test.js [hunk 2]
```diff
@@ -33,18 +33,20 @@ describe('controllers - case', () => {
         mock: true,
       };
 
+      const mockAmendments = [{
+        ukefFacilityId: '1234',
+        facilityId: '12345',
+        submittedByPim: false,
+        type: 'Cash',
```
Comment (IgnatG) [inline trade-finance-manager-ui/server/controllers/case/index.test.js:40]: do we need `type`?

File: trade-finance-manager-ui/server/controllers/case/index.test.js [hunk 3]
```diff
@@ -33,18 +33,20 @@ describe('controllers - case', () => {
...
+        status: 'In progress',
```
Comment (IgnatG) [inline trade-finance-manager-ui/server/controllers/case/index.test.js:41]: can we use a constant for this?

File: trade-finance-manager-ui/server/test-mocks/amendment-test-mocks.js (hunk 1 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/server/test-mocks/amendment-test-mocks.js:64]: done

File: trade-finance-manager-ui/server/controllers/case/index.test.js (hunk 3 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/server/controllers/case/index.test.js:41]: done

File: trade-finance-manager-ui/server/controllers/case/index.test.js (hunk 2 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/server/controllers/case/index.test.js:40]: removed

File: e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/amendment-tasks-manual.spec.js [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+    completeTask(completeTaskParams1);
+    cy.url().should('eq', relative(`/case/${dealId}/tasks`));
+    pages.tasksPage.filterRadioAllTasks().click();
+
+    pages.tasksPage.tasks.row(4, 2).status().contains('Done');
+    pages.tasksPage.tasks.row(4, 3).status().contains('To do');
+    pages.tasksPage.tasks.row(4, 4).status().contains('Cannot start yet');
+    pages.tasksPage.tasks.row(4, 5).status().contains('Cannot start yet');
+
+    // cy.login(UNDERWRITER_MANAGER_1);
```
Comment (IgnatG) [inline e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/amendment-tasks-manual.spec.js:365]: do we need this commented code?

Comment (Zainzzkk) [reply e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/amendment-tasks-manual.spec.js:365]: yes, have left todo comment.  Needed for when UW managers decision is dependent on tasks

File: trade-finance-manager-api/src/v1/controllers/amendment.controller.js [hunk 5]
```diff
@@ -47,6 +49,19 @@ const sendAmendmentEmail = async (amendmentId, facilityId) => {
...
   const payload = req.body;
+  if (payload.createTasks && payload.submittedByPim) {
+    const tasks = createAmendmentTasks(payload.requireUkefApproval);
+    payload.tasks = tasks;
+    delete payload.createTasks;
+    delete payload.requireUkefApproval;
+  }
+
+  if (payload?.taskUpdate?.updateTask) {
+    const tasks = await updateAmendmentTask(facilityId, amendmentId, payload.taskUpdate);
```
Comment (Zainzzkk) [inline trade-finance-manager-api/src/v1/controllers/amendment.controller.js:60]: not a showstopper at all: can be changed later, could this be updateAmendmentTasks instead?

Comment (IgnatG) [reply trade-finance-manager-api/src/v1/controllers/amendment.controller.js:60]: good shout! - let's do it in a separate PR though

File: trade-finance-manager-api/src/v1/controllers/deal.tasks.js [hunk 6]
```diff
@@ -24,19 +24,14 @@ const shouldCreatePartiesTask = (deal) => {
...
+  const { dealType, submissionType } = deal;
 
-  if (dealType === CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS
-    && submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIA) {
+  if (dealType === CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS && submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIA) {
     const { eligibility } = deal;
 
-    const eligibilityCriterion11 = eligibility.criteria.find((criterion) =>
-      criterion.id === 11);
+    const eligibilityCriterion11 = eligibility.criteria.find((criterion) => criterion.id === 11);
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/v1/controllers/deal.tasks.js:32]: Can we have `11` as a constant please?

**Review by abhi-markan — DISMISSED** [review]
Solid work and very comprehensive.
Few minor comments.

**Review by Zainzzkk — APPROVED** [review]
LGTM

**Pull request merged**